src/payment/strategies/paypal-commerce/paypal-commerce-payment-strategy.ts

```diff
@@ -92,13 +119,53 @@ export default class PaypalCommercePaymentStrategy implements PaymentStrategy {
         return Promise.reject(new OrderFinalizationNotRequiredError());
     }
 
-    async deinitialize(): Promise<InternalCheckoutSelectors> {
+    async deinitialize({gatewayId}: PaymentRequestOptions): Promise<InternalCheckoutSelectors> {
+        this._deinitializePollingTimer(gatewayId);
         this._orderId = undefined;
         this._paypalCommercePaymentProcessor.deinitialize();
 
         return Promise.resolve(this._store.getState());
     }
 
+    private _initializePollingMechanism(submitForm: () => void, gatewayId?: string, methodId?: any, paypalcommerce?: any ) {
+        if (gatewayId !== PaymentStrategyType.PAYPAL_COMMERCE_ALTERNATIVE_METHODS)  {
+            return;
+        }
+
+        this._pollingInterval = window.setTimeout(async () => {
+            try {
+                this._pollingTimer += POLLING_INTERVAL;
+
+                const { status } = await this._paypalCommerceRequestSender.getOrderStatus();
+
+                if (status === ORDER_STATUS_APPROVED) {
+                    this._deinitializePollingTimer(gatewayId);
+                    this._tokenizePayment({ orderID: this._paypalCommercePaymentProcessor.getOrderId() }, submitForm);
+                } else if (status === ORDER_STATUS_CREATED  && this._pollingTimer < POLLING_MAX_TIME) {
+                    this._initializePollingMechanism(submitForm, gatewayId, methodId, paypalcommerce);
+                } else {
+                    this._reinitializeButtons({ gatewayId, methodId, paypalcommerce });
+                    throw new TimeoutError();
+                }
+            } catch (e) {
+                this._deinitializePollingTimer(gatewayId);
+                paypalcommerce.onError(e);
```

Comment:
Can you add the `onError` method to the options interface https://github.com/bigcommerce/checkout-sdk-js/blob/6af83917b16e3257c19ac727368126162be0aec3/src/payment/strategies/paypal-commerce/paypal-commerce-payment-initialize-options.ts#L48
it should be optional param (we use it only for alternative methods, the regular paypal is not affected)

Reply:
`paypalcommerce.onError(e)` -> paypalcommerce.onError?.(e)

Reply:
fixed